### PR TITLE
net-p2p/deluge: version bump to 2.0.3

### PR DESCRIPTION
**Pull request by PPed72**
Closes: https://bugs.gentoo.org/688024
Signed-off-by: Paolo Pedroni <paolo.pedroni@iol.it>
Package-Manager: Portage-2.3.66, Repoman-2.3.16

### CODE REVIEW

File: net-p2p/deluge/deluge-2.0.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		dev-libs/libappindicator:3[introspection]
+		dev-python/pygobject:3[cairo,${PYTHON_USEDEP}]
+		gnome-base/librsvg
+		libnotify? ( x11-libs/libnotify[introspection] )
+		sound? ( dev-python/pygame[${PYTHON_USEDEP}] )
+	)
+	webinterface? ( dev-python/mako[${PYTHON_USEDEP}] )"
+
+python_prepare_all() {
+	local args=(
```
Comment (mgorny) [inline net-p2p/deluge/deluge-2.0.3.ebuild]: Please replace those `sed`s with a patch. This is horribly unreadable.

File: net-p2p/deluge/deluge-2.0.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	sed -i "${args[@]}" -- 'setup.py' || die
+	args=(
+		-e "s|'new_release_check': True|'new_release_check': False|"
+	)
+	sed -i "${args[@]}" -- 'deluge/core/preferencesmanager.py' || die
+
+	distutils-r1_python_prepare_all
+}
+
+esetup.py() {
```
Comment (mgorny) [inline net-p2p/deluge/deluge-2.0.3.ebuild]: Overriding eclass functions is not acceptable.

File: net-p2p/deluge/deluge-2.0.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	# Override this call from the distutils-r1 eclass.
+	# This does not respect the distutils-r1 API. DONOT copy this example.
+	set -- "${PYTHON}" setup.py "$@"
+	echo "$@"
+	"$@" || die
+}
+
+python_install_all() {
+	distutils-r1_python_install_all
+	python_export PYTHON_SITEDIR
```
Comment (mgorny) [inline net-p2p/deluge/deluge-2.0.3.ebuild]: Declare it as local variable.

File: net-p2p/deluge/deluge-2.0.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+		rm -f "${D%/}/usr/share/man/man1/deluge-web.1" || die
+	fi
+	newinitd "${FILESDIR}"/deluged.init-2 deluged
+	newconfd "${FILESDIR}"/deluged.conf-2 deluged
+	systemd_newunit "${FILESDIR}"/deluged.service-3 deluged.service
+	systemd_install_serviced "${FILESDIR}"/deluged.service.conf
+}
+
+pkg_postinst() {
+	enewgroup ${PN}
```
Comment (mgorny) [inline net-p2p/deluge/deluge-2.0.3.ebuild]: You need to replace this with GLEP 81-compliant packages, with RFC and all.

File: net-p2p/deluge/deluge-2.0.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+	SRC_URI="https://ftp.osuosl.org/pub/deluge/source/2.0/${P}.tar.xz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="GPL-3-with-openssl-exception"
+SLOT="0"
+IUSE="console geoip gtk libnotify sound webinterface"
+REQUIRED_USE="
+	${PYTHON_REQUIRED_USE}
+	sound? ( gtk )
```
Comment (mgorny) [inline net-p2p/deluge/deluge-2.0.3.ebuild]: Sort, please.

File: net-p2p/deluge/deluge-2.0.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	|| ( dev-python/slimit[${PYTHON_USEDEP}]
+		dev-lang/closure-compiler-bin )
+	dev-python/wheel[${PYTHON_USEDEP}]
+	dev-util/intltool"
+RDEPEND="net-libs/libtorrent-rasterbar[python,${PYTHON_USEDEP}]
+	dev-python/chardet[${PYTHON_USEDEP}]
+	dev-python/dbus-python[${PYTHON_USEDEP}]
+	dev-python/distro[${PYTHON_USEDEP}]
+	dev-python/rencode[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline net-p2p/deluge/deluge-2.0.3.ebuild]: Sort.

File: net-p2p/deluge/deluge-2.0.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,132 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+PYTHON_COMPAT=( python3_{5,6} )
+DISTUTILS_SINGLE_IMPL=1
+inherit distutils-r1 eutils systemd user
```
Comment (mgorny) [inline net-p2p/deluge/deluge-2.0.3.ebuild]: I don't think you're using eutils.

File: net-p2p/deluge/deluge-2.0.3.ebuild (hunk 2 above)
Comment (PPed72) [reply net-p2p/deluge/deluge-2.0.3.ebuild]: Ok. I understand.
Do you have any suggestion on how to achieve the same result. I have no idea what to do.

Comment (mgorny) [reply net-p2p/deluge/deluge-2.0.3.ebuild]: Maybe try overriding whatever you need to override via setup.cfg.

**Pull request closed without merging**